dmcontent/questions.py

```diff
@@ -391,20 +394,19 @@ def unformat_data(self, data):
         {
             "yesno-0": True,
             "yesno-1": False,
-            "evidence-0": "Yes, I did."
-            "nonDynamicListKey": 'other data'
-        }
+            "evidence-0": "Yes, I did."        }
         """
         result = {}
-        for key in data:
-            if key == self.id:
-                for question in self.questions:
-                    # For each question e.g. evidence-0, find if data exists for it and insert it into our result
-                    root, index = question.id.split('-')
-                    if root in data[self.id][int(index)]:
-                        result[question.id] = data[self.id][int(index)].get(root)
-            else:
-                result[key] = data[key]
+        data = data.get(self.id, None)
+        if not data:
+            return result
+        for question in self.questions:
+            # For each question e.g. evidence-0, find if data exists for it and insert it into our result
+            root, index = question.id.split('-')
+            question_data = data[self.id][int(index)]
+            if root in question_data:
+                result.update({question.id: question_data.get(root)})
```

Comment:
So one of the reasons I use update is to avoid the similar syntax when updating a list. Means that at a glace I can tell that result is a dict.

```
>>> x=[1, 2, 3, 4]
>>> y=2
>>> z=3
>>> x[y] = z
>>> x={1:2, 3:4}
>>> x[y]=z
# With the above x can be list or dict.
# With the below it can only be a dict
>>> x.update({y: z})
```

Another is that I like to use it like so especially in test fixtures:
```
def get_fake_brief(**kwargs):
    data = {
        'id':1,
        'lastChangedDate': '00:00:00.00000',
        'frameworkSlug': 'digital-outcomes-and-specialists'
        ...
    }
    data.update(kwargs)
    return data

brief1 = get_fake_brief()
brief2 = get_fake_brief(id=2)
brief3 = get_fake_brief(id=3, frameworkSlug='digital-outcomes-and-specialists-2')
```

Finally in multiple updates it avoids a for loop:

```
my_nonsense_dict = {}
my_updates = {'foo': 'bar', 'baz': 'quux'}
for key in my_updates:
    my_nonsense_dict[key] = my_updates[key]


my_nonsense_dict = {}
my_updates = {'foo': 'bar', 'baz': 'quux'}
my_nonsense_dict.update(my_updates)
```

So yeah it's kind of personal preference but founded on the fact that if I want to do the above things _and_ keep my code consistent then I have to use `.update` everywhere.